AK/UUID.cpp

```diff
@@ -0,0 +1,103 @@
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include <AK/Hex.h>
+#include <AK/StringBuilder.h>
+#include <AK/UUID.h>
+
+namespace AK {
+UUID::UUID()
+    : m_uuid_buffer(ByteBuffer::create_zeroed(16))
+{
+}
+
+UUID::UUID(ByteBuffer uuid_buffer)
+    : m_uuid_buffer(ByteBuffer::copy(uuid_buffer.bytes()))
+{
+    ASSERT(m_uuid_buffer.size() == 16);
+}
+
+void UUID::convert_string_view_to_uuid(const StringView& uuid_string_view)
+{
+    ASSERT(uuid_string_view.length() == 36);
+    m_uuid_buffer = ByteBuffer::create_zeroed(16);
+    auto first_unit = decode_hex(uuid_string_view.substring_view(0, 8));
+    auto second_unit = decode_hex(uuid_string_view.substring_view(9, 4));
+    auto third_unit = decode_hex(uuid_string_view.substring_view(14, 4));
+    auto fourth_unit = decode_hex(uuid_string_view.substring_view(19, 4));
+    auto five_unit = decode_hex(uuid_string_view.substring_view(24, 12));
+    m_uuid_buffer.overwrite(0, first_unit.value().data(), first_unit.value().size());
+    m_uuid_buffer.overwrite(4, second_unit.value().data(), second_unit.value().size());
+    m_uuid_buffer.overwrite(6, third_unit.value().data(), third_unit.value().size());
+    m_uuid_buffer.overwrite(8, fourth_unit.value().data(), fourth_unit.value().size());
+    m_uuid_buffer.overwrite(10, five_unit.value().data(), five_unit.value().size());
+}
+
+UUID::UUID(const StringView& uuid_string_view)
+{
+    convert_string_view_to_uuid(uuid_string_view);
+}
+
+UUID::UUID(const String& uuid_string)
+{
+    convert_string_view_to_uuid(uuid_string.substring_view(0));
+}
+
+String UUID::as_string() const
+{
+    StringBuilder builder(36);
+    builder.append(encode_hex(m_uuid_buffer.slice(0, 4)).substring_view(0));
+    builder.append('-');
+    builder.append(encode_hex(m_uuid_buffer.slice(4, 2)).substring_view(0));
+    builder.append('-');
+    builder.append(encode_hex(m_uuid_buffer.slice(6, 2)).substring_view(0));
+    builder.append('-');
+    builder.append(encode_hex(m_uuid_buffer.slice(8, 2)).substring_view(0));
+    builder.append('-');
+    builder.append(encode_hex(m_uuid_buffer.slice(10, 6)).substring_view(0));
+    return builder.to_string();
+}
+
+const ByteBuffer& UUID::as_bytebuffer() const
+{
+    return m_uuid_buffer;
+}
+
+bool UUID::operator==(const UUID& other) const
+{
+    return other.m_uuid_buffer == m_uuid_buffer;
+}
+
+bool UUID::is_null() const
+{
+    for (size_t index = 0; index < 16; index++) {
+        if (m_uuid_buffer.data()[index] != 0)
+            return false;
+    }
+    return true;
+}
```

Comment:
1. This looks like it could use `AK::all_of()` :)
2. Maybe `is_zero()` is a better name, as that's what it actually checks?

`AK::MACAddress` does both:

https://github.com/SerenityOS/serenity/blob/bbe787a0afd51307cdac2c023895c19df31519ca/AK/MACAddress.h#L80-L83

Reply:
I'll fix it right away, thanks! :)